### PR TITLE
Block aqa and tck tests if smoke tests fail or are unstable.

### LINKED ISSUE #121: Smoke tests failures didn't block the run-aqa tests as expected.
We'd like smoke tests failures would block the run-aqa tests.

Now if smoke tests job generation failed the run-aqa tests are blocked.
https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk16u/job/jdk16u-windows-x64-openj9/16/console

However if smoke tests job generate correctly but job itself failed, run-aqa test was not blocked as expected.
https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk16u/job/jdk16u-linux-x64-hotspot/15/console

Probably need to update https://github.com/adoptium/ci-jenkins-pipelines/blob/master/pipelines/build/common/openjdk_build_pipeline.groovy#L292

### PR DESCRIPTION
**Pull request by sophia-guo**
Close #121 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -1885,27 +1884,31 @@ class Build {
...
-                                platform = buildConfig.ARCHITECTURE + '_' + buildConfig.TARGET_OS
-                            }           
-                            if ( !(platform  == 'riscv64_linux' || platform =='aarch64_windows') ) {
-                                if ( !(buildConfig.JAVA_TO_BUILD == 'jdk8u' && platform == 's390x_linux') ) {
-                                    context.echo "Remote trigger Eclipse temurin AQA_Test_Pipeline job with ${platform} ${buildConfig.JAVA_TO_BUILD}"
-                                    def remoteTargets = remoteTriggerJckTests(platform, filename)
-                                    context.parallel remoteTargets
+                        //Only smoke tests succeed TCK and AQA tests will be triggerred.
+                        if (runSmokeTests() == 'SUCCESS') {
+                            // Remote trigger Eclispe Temurin JCK tests
```
Comment (smlambert) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1889]:
```suggestion
                            // Remote trigger Eclipse Temurin JCK tests
```

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -1885,27 +1884,31 @@ class Build {
...
+                            if (buildConfig.VARIANT == 'temurin' && isRelease) {
+                                def platform = ''
+                                if (buildConfig.ARCHITECTURE.contains('x64')) {
+                                    platform = 'x86-64_' + buildConfig.TARGET_OS
+                                } else {
+                                    platform = buildConfig.ARCHITECTURE + '_' + buildConfig.TARGET_OS
+                                }           
+                                if ( !(platform  == 'riscv64_linux' || platform =='aarch64_windows') ) {
+                                    if ( !(buildConfig.JAVA_TO_BUILD == 'jdk8u' && platform == 's390x_linux') ) {
+                                        context.echo "Remote trigger Eclipse temurin AQA_Test_Pipeline job with ${platform} ${buildConfig.JAVA_TO_BUILD}"
```
Comment (smlambert) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1899]:
```suggestion
                                        context.echo "Remote trigger Eclipse Temurin AQA_Test_Pipeline job with ${platform} ${buildConfig.JAVA_TO_BUILD}"
```

**Review by smlambert — APPROVED** [review]
LGTM, added some suggestions for the typos that pre-existed in the adjusted code.

**Comment by andrew-m-leonard** [comment]
@smlambert At the moment jdk8 alpineLinux Smoke tests fail due to:
```
19:17:04  /home/jenkins/workspace/build-scripts/jobs/jdk8u/jdk8u-alpine-linux-x64-temurin_SmokeTests/aqa-tests/TKG/envSettings.mk:34: *** DETECTED_JDK_VERSION value is 17, settled JDK_VERSION value is 8. JDK_VERSION value does not match. Please reset or unset JDK_VERSION.  Stop.
```
So with this change we won't be able to run tests on jdk8 Alpine ? I guess we can run manually instead?

**Comment by smlambert** [comment]
Argh!  That relates to https://github.com/adoptium/infrastructure/issues/2934 for which there is a workaround on the test side, though I kind of wanted to see a resolution via the infra containers that we are exclusively using for test.

**Comment by sophia-guo** [comment]
@smlambert @andrew-m-leonard I have updated the PR to set propagate=false, otherwise there is no way to get the downstream job status is it fails and fail the build step immediately ( we are using the catchError to avoid this now) https://issues.jenkins.io/browse/JENKINS-48475?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&showAll=true

**Review by karianna — APPROVED** [review]

**Pull request merged**